File: Modules/MooMed.Module.Search/Service/SearchService.cs
using System.Threading.Tasks;
using JetBrains.Annotations;
using MooMed.Common.Definitions.Models.Search;
using MooMed.Common.ServiceBase.Interface;

namespace MooMed.Module.Search.Service
{
    public class SearchService : ISearchService
    {
        [NotNull]
        private readonly IAccountService m_accountService;

        public SearchService([NotNull] IAccountService accountService)
        {
            m_accountService = accountService;
        }

        [ItemNotNull]
        public async Task<SearchResult> Search([NotNull] string query)
        {
            var searchResult = new SearchResult
            {
                CorrespondingUsers = await m_accountService.FindAccountsStartingWithName(query)
            };

            return searchResult;
        }
    }
}

File: Services/MooMed.Frontend/React/Modules/Search/Service/SearchService.ts
// Functionality
import { ISearchService } from "Definitions/Service";
import ModuleService from "modules/common/Service/ModuleService";
import * as searchCommunication from "modules/Search/Communication/SearchCommunication";
import { SearchResult } from "modules/Search/types";

export default class SearchService extends ModuleService implements ISearchService {

	public constructor() {
		super();
	}

	public async start() {
		
	}

	public async search(query: string): Promise<SearchResult> {
		const searchResponse = await searchCommunication.search({
			query
		});

		return {
			correspondingAccounts: searchResponse.payload.correspondingUsers
		};
	}
}